### PR TITLE
feat: add tally/invalid-json-form rule

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary

- Add `tally/invalid-json-form` rule that detects Dockerfile instructions where arguments start with `[` but contain invalid JSON (unquoted strings, single quotes, trailing commas)
- BuildKit silently falls back to shell-form for these, producing unexpected build behavior — this rule catches that at lint time
- Includes auto-fix (`FixSuggestion`, applied via `--fix-unsafe`) that repairs the JSON syntax

## Details

**Detection** (AST-level, no semantic model needed):
- Checks `node.Attributes["json"]` — absent when BuildKit's `parseJSON` failed
- Covers `CMD`, `ENTRYPOINT`, `RUN`, `SHELL`, `COPY`, `ADD`, `VOLUME`, plus `HEALTHCHECK CMD` and `ONBUILD` wrappers
- Guards against false positives: bash `[[ ]]` test syntax is excluded

**Fix patterns**:
- `CMD [bash, -lc, "echo"]` → `CMD ["bash", "-lc", "echo"]`
- `ENTRYPOINT ['/app', '--serve']` → `ENTRYPOINT ["/app", "--serve"]`
- `RUN ["echo", "hello",]` → `RUN ["echo", "hello"]`

**Files**: rule + tests (490 + 521 lines), integration fixture + 4 snapshots, docs page, index/count updates

## Test plan

- [x] Unit tests: 30+ cases covering all instruction types, HEALTHCHECK/ONBUILD, edge cases, fix validation
- [x] Integration lint case with 3-violation fixture
- [x] Integration fix cases for unquoted, single-quote, and trailing-comma patterns
- [x] `go test ./...` passes
- [x] `make lint` — 0 issues
- [x] `make cpd` — 0 duplications
- [x] Coverage for `invalid_json_form.go` ~87%

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a new tally rule (invalid-json-form) to detect malformed JSON in Dockerfile exec-form instructions (CMD, ENTRYPOINT, RUN, etc.) with suggested automatic fixes.

* **Documentation**
  * Updated rule documentation and reference pages to include the new rule and adjusted supported-rule counts.

### CODE REVIEW

File: internal/rules/tally/invalid_json_form_test.go [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+			Content:        "FROM alpine:3.20\nONBUILD WORKDIR /app\n",
+			WantViolations: 0,
+		},
+		{
+			Name:           "HEALTHCHECK NONE — no violation",
+			Content:        "FROM alpine:3.20\nHEALTHCHECK NONE\n",
+			WantViolations: 0,
+		},
+	})
+}
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/tally/invalid_json_form_test.go]: ✅ Addressed in e0612be. Thanks @coderabbitai!

All three suggestions applied:
1. **POSIX `[ test ]` false positive fixed** — added guard: if content between `[` and `]` has no commas and starts with `[ ` (space), it's a shell test expression, not JSON
2. **ADD unquoted** test case added
3. **COPY unquoted** test case added

File: internal/rules/tally/invalid_json_form.go [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+		}
+	}
+
+	// Add the last element.
+	if current.Len() > 0 || len(parts) > 0 {
+		parts = append(parts, current.String())
+	}
+
+	return parts
+}
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/tally/invalid_json_form.go]: ✅ Addressed in 23a54c1. Thanks @coderabbitai!

Converted `splitJSONElements` from byte-level (`s[i]`) to rune-level (`[]rune(s)`) iteration so multi-byte UTF-8 characters inside quoted strings are handled correctly.

File: internal/rules/tally/invalid_json_form_test.go [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+			got, ok := tryRepairJSON(tt.input)
+			if ok != tt.wantOK {
+				t.Fatalf("tryRepairJSON(%q) ok = %v, want %v", tt.input, ok, tt.wantOK)
+			}
+			if ok && got != tt.want {
+				t.Errorf("tryRepairJSON(%q) = %q, want %q", tt.input, got, tt.want)
+			}
+		})
+	}
+}
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/tally/invalid_json_form_test.go]: ✅ Addressed in 3c829cb. Thanks @coderabbitai!

Added all three adversarial test cases:
- `['say "hello"']` → `["say \"hello\""]` (internal double quotes)
- `[C:\path]` → `["C:\\path"]` (backslash escaping)
- `["", "b"]` → `["", "b"]` (empty string element preserved)

Also added matching `ensureDoubleQuoted` unit tests for the same edge cases.

File: internal/rules/tally/invalid_json_form.go [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+	for strings.HasPrefix(rest, "--") {
+		end := strings.IndexFunc(rest, unicode.IsSpace)
+		if end < 0 {
+			return ""
+		}
+		rest = strings.TrimSpace(rest[end:])
+	}
+
+	return rest
+}
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/tally/invalid_json_form.go]: ✅ Addressed in 20df511. Thanks @coderabbitai!

`extractOnbuildArgs` now skips past "ONBUILD" before searching for the sub-instruction keyword, preventing false matches if the keyword happens to appear earlier in the line.
